ocs_ci/ocs/cluster.py

```diff
@@ -912,6 +912,74 @@ def get_osd_utilization():
     return osd_filled
 
 
+def get_ceph_df_detail():
+    """
+    Get ceph osd df detail
+    :return: dict returned from cmd
+    """
+    ceph_cmd="ceph df detail"
+    ct_pod = pod.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+    return output
+
+
+def validate_replica_data(pool_name, replica):
+    """
+    Check if data is replica 2 or 3
+    :param int replica: size of the replica(2,3)
+    :param string pool_name: name of the pool to check replica
+    :return: true if replica is verified, false if not
+    """
+    from ocs_ci.ocs.exceptions import PoolNotFound
+    output = get_ceph_df_detail()
+    for pool in output.get('pools'):
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            stored = pool.get('stats').get('stored')
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').get('compress_under_bytes')
+            byte_used = byte_used + compress_under_bytes - compress_bytes_used
+            store_ratio = byte_used / stored
+            if (replica+0.2) > store_ratio > (replica-0.2):
+                logger.info(f"pool {pool_name} meet rep {replica} size")
+                return True
+            else:
+                logger.info(f"pool {pool_name} meet do not meet rep {replica} size"
+                            f"Store ratio is {store_ratio}")
+                return False
+        if pool == output.get('pools')[-1]:
```

Comment:
What does `pool == output.get('pools')[-1]` condition mean? It would be more readable, if we just have a boolean variable about whether we found the pool with poo_name, and then later here checked if that is the case and raise an exception accordingly.

Reply:
It means if the pool = the last pool in the list. Why we have to keep another variable to track this after the loop when I can handle it with a simple statement. I can simplify before the loop the expression like:
```pool_list = output.get('pools')```
and then call
```if pool == pool_list[-1]```
will be much more readable
wdyt?

Reply:
Change the code to my suggestion. Not resolving the issue till your thoughts on it.

Reply:
Actually, the most readable and simple would be to just raise an exception without the if statement. Code after the if won't be executed if we found the pool.

Reply:
But the loop have  states that it doesn't find the pool because it is iterating over a list to find the pool and most of the time the loop have state which it doesn't find the pool till it is found, so how can we drop the if statement?

Reply:
Finally understood. Remove if and moved it out of the loop